File: LinkedList/LinkedList.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


class LinkedList:
    def __init__(self):
        self.head = None

    def printList(self):
        temp = self.head
        while(temp):
            print(temp.data)
            temp = temp.next

    def add(self, data):
        temp = self.head
        self.head = Node(data)
        self.head.next = temp


llist = LinkedList()
llist.head = Node(1)

second = Node(2)


llist.head.next = second
second.next = None

llist.add(3)
llist.add(4)
llist.add(5)
llist.printList()

File: LinkedList/LinkedList.php
<?php

class Node
{
    public $data;
    public $next = NULL;
    function __construct($data)
    {
        $this->data = $data;
    }
}

class LinkedList
{
    public $head;
    function __construct()
    {
        $this->head = NULL;
    }
    function printList()
    {
        $temp = &$this->head;
        while ($temp) {
            echo $temp->data;
            echo " -> ";
            $temp = &$temp->next;
        }
    }
    function add($data)
    {
        $temp = $this->head;
        $this->head = new Node($data);
        $this->head->next = $temp;
    }
}
$list = new LinkedList();

$list->add(1);
$list->add(2);
$list->add(3);
$list->add(4);



$list->printList();
